### ISSUE TITLE
table behavior within quarto books

### ISSUE THREAD

**Issue by topepo** [rstudio/gt#1085]
I've found two features that have some issues when gt 0.7.0 used inside of quarto (1.2.233). 

First, the table width settings via `tab_options()` don't seen to work. There is a reprex in [this repo](https://github.com/topepo/gt-quarto-test). In the `width.qmd` file, using [`tab_options(table.width = px(40))`](https://github.com/topepo/gt-quarto-test/blob/main/width.qmd#L11) or [`tab_options(table.width = pct(80))`](https://github.com/topepo/gt-quarto-test/blob/main/width.qmd#L22) don't seem to affect the table width. 

The second issue is the cell background in a column spanner. [This code](https://github.com/topepo/gt-quarto-test/blob/main/colors.qmd#L7:L12) produces the spanner column but the color doesn't automatically carry through: 

<img width="771" alt="image" src="https://user-images.githubusercontent.com/5731043/197806047-522e328f-f656-4a0b-a391-ab672d87521a.png">

**Comment by cscheid** [comment]
This is a quarto bug! We'll fix it on our end. Sorry, @topepo!

Turns out that quarto overrides some of gt's default style in order to match the standard themes we have. This has two consequences:

1. we're missing an override for the multiple colspan entries.
2. we include a `table { width: 100% }` rule to match the style of other quarto tables. You can override this yourself by adding a css rule of higher specificity than this one: https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/html/_quarto-rules.scss#L515

We'll address 1) in a fix right away. But for 2) you'll need to do some of the work (since the 100% width behavior is by design on our part.)

**Comment by topepo** [comment]
Thanks. 

Should I find a way to add a subclass to the output of gt tables (and write css for that) or would a scss file be able to override the css for `table.gt_table`? Not my forte here so ELI5 :-)

Was the hardcoding off 100% width a default of convenience or is there some more specific reason? TBH it doesn't seem like a good default.

**Comment by cscheid** [comment]
> Should I find a way to add a subclass to the output of gt tables (and write css for that) or would a scss file be able to override the css for table.gt_table? Not my forte here so ELI5 :-)

The latter. The usual CSS trick is to find a way to get a selector of higher specificity. The easiest way to do it is to wrap the table with a div that has either an id or a custom class, and then write a selector that picks both. So that would be something like

````

::: my-class

```{r}
# here's the gt code
```

:::

````

and then adding CSS that goes

```css
div.my-class table.gt_table {
    width: inherit; // or whatever you want
}
```

This should work (though I haven't tested it so I might have details wrong!)

> Was the hardcoding off 100% width a default of convenience or is there some more specific reason? TBH it doesn't seem like a good default.

Consistency: it matches other table styles in quarto.
